esx_service/utils/auth_api.py

```diff
@@ -196,6 +211,29 @@ def _tenant_create(name, description, default_datastore, default_privileges, vm_
     
     return None, tenant
 
+def _tenant_modify(name, new_name=None, description=None):
+    """ API to modify a tenant """
+    error_info, tenant = get_tenant_from_db(name)
+    if error_info:
+        return error_info
+    
+    if not tenant:
+        error_info = error_code.TENANT_NOT_EXIST.format(name)
+        return error_info
+    
+    error_info, auth_mgr = get_auth_mgr()
+    if error_info:
+        return error_info
+    if new_name:    
+        error_info = tenant.set_name(auth_mgr.conn, name, new_name)
```

Comment:
It is true the _tenant_update API user doesn't need to worry about the connection. However, in my opinion the user of DockerVolumeTenant.set_name() API should not worry about the connection either. Basically we should isolate the DB operation to any upper-level business logic.

Anyway, I agree this will need a lot of code refactoring. We can address this later.